### ISSUE TITLE
"--listen-address" argument doesn't support hostname as the help message describes

### ISSUE THREAD

**Issue by kongove** [scylladb/scylla#1310]
_Installation details_
Scylla version (or git commit hash):57413618e80e2c9b59184756be1f2c3ccc87bf14 (Thu Jun 2 21:30:10 2016 +0200)
Cluster size: single node
OS (RHEL/CentOS/Ubuntu/AWS AMI): Fedora 23
### Description:

IP address (192.168.31.245) was map to hostname 'node0' by /etc/hosts, the IP was assigned to local interface by DHCP. But I failed to launch scylladb when I assigned listen-address by hostname 'node0'

```
[@MiWiFi-R1D-srv scylla]# ifconfig |grep inet
        inet 192.168.31.245  netmask 255.255.255.0  broadcast 192.168.31.255
```

[@MiWiFi-R1D-srv scylla]# ./build/release/scylla --datadir tmp --commitlog-directory tmp --smp 1 --max-io-requests 100 --developer-mode 1 --listen-address node0 
Scylla version 666.development-20160605.5741361 starting ...
WARN  2016-06-05 17:47:06,684 [shard 0] config - Option partitioner is not (yet) used.
ERROR 2016-06-05 17:47:06,685 [shard 0] init - **Bad configuration: invalid 'listen_address': node0: std::runtime_error (Wrong format for IPv4 address node0.** Please ensure it's in dotted-decimal format)
Exiting on unhandled exception of type 'bad_configuration_error': std::exception
### Additional help message:

```
  --listen-address arg (=localhost)     The IP address or hostname that 
                                    Cassandra binds to for connecting to 
                                    other Cassandra nodes. Set this 
                                    parameter or listen_interface, not 
                                    both. You must change the default 
                                    setting for multiple nodes to 
                                    communicate:

                                    Generally set to empty. If the node is 
                                    properly configured (host name, name 
                                    resolution, and so on), Cassandra uses 
                                    InetAddress.getLocalHost() to get the 
                                    local address from the system.
                                    For a single node cluster, you can use 
                                    the default setting (localhost).
                                    If Cassandra can't find the correct 
                                    address, you must specify the IP 
                                    address or host name.
                                    Never specify 0.0.0.0; it is always 
                                    wrong.

  --rpc-address arg (=localhost)        The listen address for client 
                                    connections (Thrift RPC service and 
                                    native transport).Valid values are:

                                    unset:   Resolves the address using the
                                    hostname configuration of the node. If 
                                    left unset, the hostname must resolve 
                                    to the IP address of this node using 
                                    /etc/hostname, /etc/hosts, or DNS.
                                    0.0.0.0 : Listens on all configured 
                                    interfaces, but you must set the 
                                    broadcast_rpc_address to a value other 
                                    than 0.0.0.0.
                                    IP address
                                    hostname
                                    Related information: Network
```

**Comment by penberg** [comment]
@kongove It's indeed a bug. The help text is misleading as Scylla does not resolve host names and expects an IP address. We should fix that.

**Comment by penberg** [comment]
Another issue that's related to listen address configuration:

https://github.com/scylladb/scylla/issues/1058

**Comment by avikivity** [comment]
Problem is that seastar doesn't have a DNS resolver. We can't use the system resolver because it's synchronous.  Options are to use the syscall thread for that, or to integrate c-ares.

**Comment by slivne** [comment]
@elcallio as a start we can send a patch that we support only ip address

**Comment by tzach** [comment]
@slivne I guess this will not be a part of 1.5

**Comment by elcallio** [comment]
This was fixed by feffc2bbe1a219b31787e9ca275e00ab3cf8cbbf
